stdlib/public/Darwin/Foundation/NSValue.swift.gyb

```diff
@@ -23,3 +23,16 @@ ${ ObjectiveCBridgeableImplementationForNSValue("CGVector") }
 ${ ObjectiveCBridgeableImplementationForNSValue("CGSize") }
 ${ ObjectiveCBridgeableImplementationForNSValue("CGAffineTransform") }
 
+extension NSValue {
+    @_transparent // this must be transparent to pass the type encoding over to the _getObjCTypeEncoding function
+    public func value<StoredType>(of type: StoredType.Type) -> StoredType? {
+        guard strcmp(objCType, _getObjCTypeEncoding(StoredType.self)) == 0 else {
```

Comment:
`@transparent` doesn't guarantee this, but I'm also concerned about adding this check to NSValue when it wasn't there before. You can see #19691 for an example where this would cause a problem.

Reply:
Well there are some other monsters under the water here. Passing in String.self or Any.self crashes the compiler.